### PR TITLE
Add myself to bootstrap reviewers

### PR DESCRIPTION
**Pull request by jyn514**
Forgot that I had to do this manually :see_no_evil: @Mark-Simulacrum this should take some of the review burden off you.

I also deleted a few configs that haven't done anything for ages.

### CODE REVIEW

**Comment by jyn514** [comment]
BTW, this removes the `src/doc` reviews, since the doc team doesn't exist anymore - @ehuss are you interested in being a reviewer? or do you know who normally reviews those PRs?

File: highfive/configs/rust-lang/rust.json [hunk 1]
```diff
@@ -111,24 +115,22 @@
...
-        "src/bootstrap":                         ["@Mark-Simulacrum"],
-        "src/build_helper":                      ["@Mark-Simulacrum"],
+        "src/bootstrap":                         ["bootstrap"],
         "src/ci":                                ["infra-ci"],
-        "src/doc":                               ["doc"],
         "src/etc":                               ["@Mark-Simulacrum"],
         "src/librustdoc":                        ["rustdoc"],
         "src/llvm-project":                      ["@cuviper"],
-        "src/stage0.txt":                        ["@Mark-Simulacrum"],
+        "src/stage0.txt":                        ["bootstrap"],
```
Comment (Mark-Simulacrum) [inline highfive/configs/rust-lang/rust.json:123]: This is .json now, I think.

Comment (jyn514) [reply highfive/configs/rust-lang/rust.json]: Yes it is, good catch. Fixed.

**Comment by ehuss** [comment]
I think @GuillaumeGomez tended to do docs reviews.  I opened #411 for an alternative.

**Pull request merged**